test/vault_auth_test.go

```diff
@@ -1,25 +1,93 @@
 package test
 
 import (
+	"fmt"
 	"testing"
+	"time"
+
+	"github.com/gruntwork-io/terratest/modules/http-helper"
+	"github.com/gruntwork-io/terratest/modules/random"
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/gruntwork-io/terratest/modules/test-structure"
 )
 
-func TestVaultEC2AuthWithUbuntuAmi(t *testing.T) {
-	t.Parallel()
-	runVaultEC2AuthTest(t, "ubuntu16-ami")
-}
+const VAULT_EC2_AUTH_PATH = "examples/vault-ec2-auth"
+const VAULT_IAM_AUTH_PATH = "examples/vault-iam-auth"
+
+const VAR_VAULT_AUTH_SERVER_NAME = "auth_server_name"
+const VAR_VAULT_SECRET_NAME = "example_secret"
+const VAR_VAULT_IAM_AUTH_ROLE = "example_role_name"
+
+const OUTPUT_AUTH_CLIENT_IP = "auth_client_public_ip"
+
+// Test the Vault EC2 authentication example by:
+//
+// 1. Copying the code in this repo to a temp folder so tests on the Terraform code can run in parallel without the
+//    state files overwriting each other.
+// 2. Building the AMI in the vault-consul-ami example with the given build name
+// 3. Deploying that AMI using the example Terraform code setting an example secret
+// 4. Waiting for Vault to boot, then unsealing the server, creating a Vault Role to allow logins from instances with a specific EC2 property and writing the example secret
+// 5. Waiting for the client to login, read the secret and launch a simple web server with the contents read
+// 6. Making a request to the webserver started by the auth client
+func runVaultEC2AuthTest(t *testing.T, amiId string, sshUserName string) {
```

Comment:
Nothing changed here besides removing the step of building the ami. Everything shows as new because I moved it from file. All tests were on `vault_helper` and with time that had become a monstrosity